libs/vkd3d/resource.c

```diff
@@ -2389,15 +2391,25 @@ static void d3d12_resource_destroy(struct d3d12_resource *resource, struct d3d12
 static HRESULT d3d12_resource_create_vk_resource(struct d3d12_resource *resource, struct d3d12_device *device)
 {
     const D3D12_HEAP_PROPERTIES *heap_properties;
+    D3D12_RESOURCE_DESC padded_desc;
     HRESULT hr;
 
     heap_properties = resource->flags & VKD3D_RESOURCE_RESERVED
         ? NULL : &resource->heap_properties;
 
     if (resource->desc.Dimension == D3D12_RESOURCE_DIMENSION_BUFFER)
     {
+        padded_desc = resource->desc;
+
+        /* If we attempt to bind sparse buffer with non-64k pages, we crash drivers.
+         * Specs seems a bit unclear how non-aligned VkBuffer sizes are supposed to work,
+         * so be safe. Pad out sparse buffers to their natural page size. */
+        if (resource->desc.Dimension == D3D12_RESOURCE_DIMENSION_BUFFER &&
+                (resource->flags & VKD3D_RESOURCE_RESERVED))
+            padded_desc.Width = align64(padded_desc.Width, VKD3D_TILE_SIZE);
```

Comment:
Can we handle this in vkd3d_create_buffer instead? We do some stuff regarding sparse resources there already, I think that would be the better place.